### ISSUE TITLE
error:09091064:PEM routines:PEM_read_bio_ex:bad base64 decode

### ISSUE THREAD

**Issue by TLKG** [auth0/node-jsonwebtoken#816]
### Description
Token validated ok at jwt.io. Received error with `.verify`

Error:
```
code:'ERR_OSSL_PEM_BAD_BASE64_DECODE'
function:'PEM_read_bio_ex'
library:'PEM routines'
reason:'bad base64 decode'
message:'error:09091064:PEM routines:PEM_read_bio_ex:bad base64 decode'
stack:'Error: error:09091064:PEM routines:PEM_read_bio_ex:bad base64 decode\n    at Verify.verify (internal/crypto/sig.js:188:24)
```

> Provide a clear and concise description of the issue, including what you expected to happen.

### Reproduction

```
const jwt = require("jsonwebtoken");
var vtoken = "...";  // validated by jwt.io
var key = "...";  // value copied from jwt.io
var publickey = "-----BEGIN CERTIFICATE-----\n" + key + "\n-----END CERTIFICATE-----";  
var payload= jwt.verify(vtoken, publickey, { algorithms: ['RS256'] });
```
> - Screenshots
![image](https://user-images.githubusercontent.com/22269729/157336722-b620e14c-268a-4a3d-8d3c-c4f14109a7e4.png)

### Environment

> Please provide the following:

- **8.5.1**
- VSC version: 1.63.1 (system setup)
Commit: fe719cd3e5825bf14e14182fddeb88ee8daf044f
Date: 2021-12-14T02:13:54.292Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19044

**Comment by elitan** [comment]
@TLKG Was this fixed? If yes, how?

**Comment by TLKG** [comment]
@elitan Following works locally

async function getMSPublicKey(misc)
{
    var vurl = "https://login.microsoftonline.com/" + misc.tenantId + "/v2.0/.well-known/openid-configuration";
    const x1 = await fetch(vurl);
    const x2 = await x1.json();
    const x3 = await fetch(x2.jwks_uri);
    const k = await x3.json();
    return pkey = k.keys.find( k => k.kid === misc.kid).x5c[0];
}
var publickey = "-----BEGIN CERTIFICATE-----\n" + await getMSPublicKey(vmisc) + "\n-----END CERTIFICATE-----";
var payload = jwt.verify(theToken, publickey, { algorithms: ['RS256'] });

After deployed to AZ got ["JsonWebTokenError: invalid algorithm"](https://github.com/auth0/node-jsonwebtoken/issues/820).

**Comment by elitan** [comment]
Thanks. The issue I had was an incorrect key string. It was solved on my end. Thanks.

**Comment by eriegz** [comment]
Should this issue get closed then?

**Comment by Yudota** [comment]
try this:

async function getMSPublicKey(misc)
{
    var vurl = "https://login.microsoftonline.com/" + misc.tenantId + "/v2.0/.well-known/openid-configuration";
    const x1 = await fetch(vurl);
    const x2 = await x1.json();
    const x3 = await fetch(x2.jwks_uri);
    const k = await x3.json();
    return pkey = k.keys.find( k => k.kid === misc.kid).x5c[0];
}
var publickey = "-----BEGIN CERTIFICATE-----\n" + await getMSPublicKey(vmisc) + "\n-----END CERTIFICATE-----\n";
var payload = jwt.verify(theToken, publickey, { algorithms: 'RS256' });

**Comment by copremesis** [comment]
Yeah in my case using a different library it was essentially a copy & paste error with the key
